seahub/share/models.py

```diff
@@ -150,6 +152,79 @@ def get_dir_link_by_path(self, username, repo_id, path):
     def get_valid_dir_link_by_token(self, token):
         return self._get_valid_file_share_by_token(token)
 
+
+class ExtraSharePermissionManager(models.Manager):
+    def get_user_permission(self, repo_id, username):
+        """Get user permission  in Library.
+        return
+            e.g. 'admin'
+        """
+        record_list = super(ExtraSharePermissionManager, self).filter(
+            repo_id=repo_id, share_to=username
+        )
+        if len(record_list) > 0:
+            return record_list[0].permission
+        else:
+            return None
+
+    def get_repos_with_admin_share_to(self, username):
+        """Get repo id with the admin permission record.
```

Comment:
Get repo id with the admin permission record.  -> Get repo id list a user has admin permission.